File: SampleCode/VisaAPICalls/Ruby/VisaDirectSample.rb
require 'rest-client'
require 'json'

# Sample post body payload
$body='''{
  "businessApplicationId": "AA",
  "merchantCategoryCode": 6012,
  "pointOfServiceCapability": {
    "posTerminalType": "4",
    "posTerminalEntryCapability": "2"
  },
  "feeProgramIndicator": "123",
  "systemsTraceAuditNumber": 300259,
  "retrievalReferenceNumber": "407509300259",
  "foreignExchangeFeeTransaction": "10.00",
  "cardAcceptor": {
    "name": "Acceptor 1",
    "terminalId": "365539",
    "idCode": "VMT200911026070",
    "address": {
      "state": "CA",
      "county": "081",
      "country": "USA",
      "zipCode": "94404"
    }
  },
  "magneticStripeData": {
    "track1Data": "1010101010101010101010101010"
  },
  "senderPrimaryAccountNumber": "4005520000011126",
  "senderCurrencyCode": "USD",
  "surcharge": "2.00",
  "localTransactionDateTime": "2021-10-26T21:32:52",
  "senderCardExpiryDate": "2013-03",
  "pinData": {
    "pinDataBlock": "1cd948f2b961b682",
    "securityRelatedControlInfo": {
      "pinBlockFormatCode": 1,
      "zoneKeyIndex": 1
    }
  },
  "cavv": "0000010926000071934977253000000000000000",
  "pointOfServiceData": {
    "panEntryMode": "90",
    "posConditionCode": "0",
    "motoECIIndicator": "0"
  },
  "acquiringBin": 409999,
  "acquirerCountryCode": "101",
  "amount": "112.00"
}'''

def test_pull_funds
  base_uri = 'visadirect/'
  resource_path = 'fundstransfer/v1/pullfundstransactions/'
  url = 'https://sandbox.api.visa.com/' + base_uri + resource_path
  user_id = 'put your user id for the app from VDP Portal here'
  password = 'put your password for the app from VDP Portal here'
  key_path = 'put the private key pem file path here'
  cert_path = 'put the client certificate pem file path here'
  headers = {'content-type'=> 'application/json', 'accept'=> 'application/json'}
  begin
    response = RestClient::Request.execute(
        :method => :post,
        :url => url,
        :headers => headers,
        :payload => $body,
        :user => user_id, :password => password,
        :ssl_client_key => OpenSSL::PKey::RSA.new(File.read(key_path)),
        :ssl_client_cert =>  OpenSSL::X509::Certificate.new(File.read(cert_path))
    )
  rescue RestClient::ExceptionWithResponse => e
    response = e.response
  end
  return response
end

puts JSON.pretty_generate(JSON.parse(test_pull_funds()))

File: SampleCode/VisaAPICalls/Node/VisaDirectSample.js
var request = require('request');
var fs = require('fs');

var data = JSON.stringify({
    "systemsTraceAuditNumber" : "451001",
    "retrievalReferenceNumber" : "330000550000",
    "localTransactionDateTime" : "2016-01-21T13:32:01",
    "acquiringBin" : "408999",
    "acquirerCountryCode" : "840",
    "senderPrimaryAccountNumber" : "4895142232120006",
    "senderCardExpiryDate" : "2015-10",
    "senderCurrencyCode" : "USD",
    "amount" : "124.02",
    "businessApplicationId" : "AA",
    "surcharge" : "11.99",
    "foreignExchangeFeeTransaction" : "11.99",
    "cavv" : "0700100038238906000013405823891061668252",
    "cardAcceptor" : {
        "name" : "Visa Inc. USA-Foster City",
        "terminalId" : "ABCD1234",
        "idCode" : "ABCD1234ABCD123",
        "address" : {
            "state" : "CA",
            "county" : "San Mateo",
            "country" : "USA",
            "zipCode" : "94404"
        }
    }
});

var req = request.defaults();

var userId = 'put your user id for the app from VDP Portal here' ;
var password = 'put your password for the app from VDP Portal here';
var keyFile = 'put the private key pem file path here';
var certificateFile ='put the client certificate pem file path here';

req.post({
    uri : "https://sandbox.api.visa.com/visadirect/fundstransfer/v1/pullfundstransactions",
    key: fs.readFileSync(keyFile),  
    cert: fs.readFileSync(certificateFile),
    headers: {
      'Content-Type' : 'application/json',
      'Accept' : 'application/json',
      'Authorization' : 'Basic ' + new Buffer(userId + ':' + password).toString('base64')
    },
    body: data
  }, function(error, response, body) {
    if (!error) {
      console.log("Response Code: " + response.statusCode);
      console.log("Headers:");
      for(var item in response.headers) {
        console.log(item + ": " + response.headers[item]);
      }
      console.log("Body: "+ body);
    } else {
      console.log("Got error: " + error.message);
    }
  }
);  